### PR TITLE
fix: add prometheus metric monitor

### PR DESCRIPTION
**Pull request by fpChan**

### CODE REVIEW

File: offchain-modules/packages/x/src/monitor/rpc-metric.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  public rpcRequestDurationms: Prometheus.Histogram<any>;
+  private gateway: Prometheus.Pushgateway;
+  private openPushMetric: boolean;
+  constructor(pushGatewayURL: string) {
+    const register = new Prometheus.Registry();
+    this.rpcRequestDurationms = new Prometheus.Histogram({
+      name: 'rpc_request_duration_ms',
+      help: 'Duration of rpc server requests in ms',
+      labelNames: ['method', 'status'],
+      buckets: [0.1, 5, 15, 50, 100], // buckets for response time from 0.1ms to 100ms
```
Comment (homura) [inline offchain-modules/packages/x/src/monitor/rpc-metric.ts:16]: IMHO, I'dont think the API like `generateXTransaction` could resolved in 0.1ms - 100ms, maybe we can use the [default buckets](https://github.com/siimon/prom-client/blob/c7b9a9ddd048f6a1921fa1b33c56a5296713aa2b/lib/histogram.js#L15) from prom-client recommended, which is `[0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1, 2.5, 5, 10]`

File: offchain-modules/packages/x/src/monitor/relayer-metric.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  private static _relayBridgeTxNum: Prometheus.Counter<any>;
+  private static _relayBridgeTokenAmountNum: Prometheus.Gauge<any>;
+
+  private static _gateway: Prometheus.Pushgateway;
+  private static _openPushMetric: boolean;
+
+  init(role: forceBridgeRole, pushGatewayURL: string): void {
+    const register = new Prometheus.Registry();
+
+    RelayerMetric._relayBlockHeightNum = new Prometheus.Gauge({
```
Comment (homura) [inline offchain-modules/packages/x/src/monitor/relayer-metric.ts:24]: `static init(pushGatewayURL: string)`
`constructor(role: Role)`

we should avoid change the static items in an instance method, it's easy to be confused

File: offchain-modules/packages/x/src/monitor/rpc-metric.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (fpChan) [reply offchain-modules/packages/x/src/monitor/rpc-metric.ts:16]: // getBridgeTransactionSummaries need 0～5ms. maybe more.
    // getAssetList need 0～5ms
    // getBalance  need 5 ～15ms，maybe more
    // generateBridgeInNervosTransaction more than 500 ms which I ignored.

File: offchain-modules/packages/x/src/monitor/relayer-metric.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      name: `${role}_block_height_number`,
```
[the remark this answers is not in the document]
Comment (fpChan) [reply offchain-modules/packages/x/src/monitor/relayer-metric.ts:23]: the roles prefix is used to do this

File: offchain-modules/packages/app-multisign-server/src/sigServer.ts [hunk 4]
```diff
@@ -90,17 +94,30 @@ export async function startSigServer(config: Config, port: number) {
...
   app.post(apiPath, (req, res) => {
     logger.info('request', req.method, req.body);
+    const startTime = Date.now();
     const jsonRPCRequest = req.body;
     // server.receive takes a JSON-RPC request and returns a promise of a JSON-RPC response.
     // Alternatively, you can use server.receiveJSON, which takes JSON string as is (in this case req.body).
     server.receive(jsonRPCRequest).then((jsonRPCResponse) => {
       if (jsonRPCResponse) {
         res.json(jsonRPCResponse);
+        SigServer.metrics.setSigServerRequestMetric(
```
Comment (solargatsby) [inline offchain-modules/packages/app-multisign-server/src/sigServer.ts:104]: add a count metric  in both success and failed response

File: offchain-modules/packages/app-rpc-server/src/server.ts [hunk 5]
```diff
@@ -77,17 +80,20 @@ export async function startRpcServer(config: Config): Promise<void> {
...
     logger.info('request', req.method, req.body);
     const jsonRPCRequest = req.body;
+    const startTime = Date.now();
     // server.receive takes a JSON-RPC request and returns a promise of a JSON-RPC response.
     // Alternatively, you can use server.receiveJSON, which takes JSON string as is (in this case req.body).
     void server.receive(jsonRPCRequest).then((jsonRPCResponse) => {
       if (jsonRPCResponse) {
         res.setHeader('Content-Type', 'application/json; charset=utf-8');
         res.json(jsonRPCResponse);
+        metrics.setRpcRequestDurationMetric(jsonRPCRequest.method, 'success', Date.now() - startTime);
```
Comment (solargatsby) [inline offchain-modules/packages/app-rpc-server/src/server.ts:90]: add a count metric in both success and failed response

**Pull request merged**